src/fortran_routing/mc_pylink_v00/Reservoir_singleTS/module_levelpool.f90

```diff
@@ -71,27 +73,31 @@ real :: dh, dh1, dh2, dh3      ! Depth in weir, and height function for 3 order
         !!DJG Set hardwire for LAKE_OPT...move specification of this to namelist in
         !future versions...
         LAKE_OPT = 2
-        Htmp = H0  !temporary set of incoming lake water elevation...
-        !hdiff_vol = 0.0
-        !qdiff_vol = 0.0
+        exp32 = 3._prec/2._prec
+        quot23 = 0.667_prec
+        zero = 0.0_prec
+        gravity = 9.81_prec
+        !hdiff_vol = zero
+        !qdiff_vol = zero
 
         !!DJG IF-block for lake model option  1 - outflow=inflow, 2 - Chow et al level
         !pool, .....
         if (LAKE_OPT == 1) then     ! If-block for simple pass through scheme....
 
            qo1 = qi1                 ! Set outflow equal to inflow at current time
-           H = Htmp                  ! Set new lake water elevation to incoming lake el.
+           H = H0                    ! Set new lake water elevation to incoming lake el.
 
         else if (LAKE_OPT == 2) then   ! If-block for Chow et al level pool scheme
 
+           H = H0                    ! Set new lake water elevation to incoming lake el.
```

Comment:
This was the critical change. Without this, the depth never comes into the equation.